### PR TITLE
feat: document different AI backends

### LINKED ISSUE #22: feature: Document different AI backends
Checklist:
* [x] I've searched for similar issues and couldn't find anything matching
* [ ] I've discussed this feature in the #k8sgpt slack channel

## Is this feature request related to a problem?
* [ ] Yes
* [x] No

## Description 

It is now possible to use different AI backends e.g. https://github.com/k8sgpt-ai/k8sgpt/pull/309
This is not yet reflected in the documentation

### PR DESCRIPTION
**Pull request by rakshitgondwal**
Closes #22

## 📑 Description
Added documentation for different AI backends supported by K8sGPT, which are:
- OpenAI
- Azure OpenAI
- LocalAI
- FakeAI

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: docs/reference/providers/backend.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# K8sGPT AI Backends
+
+A Backend or a Provider is a service that provides access to the AI language model. There are many different backends available for K8sGPT. Each backend has its own strengths and weaknesses, so it is important to choose the one that is right for your needs.
+
+Currently we have a total of 4 backends available:
+- OpenAI
```
Comment (AnaisUrlichs) [inline docs/reference/providers/backend.md:6]: Could you please add the links to the main "website" for each backend?

File: docs/reference/providers/backend.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+- Azure OpenAI
+- LocalAI
+- FakeAI
+
+## OpenAI
+
+OpenAI is the default backend for K8sGPT. OpenAI comes with the access to powerful language models such as GPT-3.5-Turbo, GPT-4. If you are looking for a powerful and easy-to-use language modeling service, OpenAI is a great option.
```
Comment (AnaisUrlichs) [inline docs/reference/providers/backend.md:13]: @AlexsJones @thschue Do we want to recommend people e.g. who are just getting started with K8sGPT and are not familiar with the different backends to use OpenAI as a default?

**Review by AnaisUrlichs — COMMENTED** [review]
I made some comments but this looks great, thank you

File: docs/reference/providers/backend.md (hunk 2 above)
Comment (AlexsJones) [reply docs/reference/providers/backend.md:13]: I would probably recommend OpenAI as the default yeah.

File: docs/reference/providers/backend.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    k8sgpt auth --backend azureopenai --baseurl https://<your Azure OpenAI endpoint> --engine <deployment_name> --model <model_name>
+    ```
+- Now you are ready to analyze with the Azure OpenAI backend:
+    ```
+    k8sgpt analyze --explain --backend azureopenai
+    ```
+
+## LocalAI
+
+LocalAI is a local model, which is an OpenAI compatible API. It uses llama.cpp and ggml to run inference on consumer-grade hardware. Models supported by LocalAI for instance are Vicuna, Alpaca, LLaMA, Cerebras, GPT4ALL, GPT4ALL-J and koala.
```
[the remark this answers is not in the document]
Comment (rakshitgondwal) [reply docs/reference/providers/backend.md:46]: I didn't pick up anything from their website, but I modified this [piece](https://github.com/k8sgpt-ai/k8sgpt#running-local-models) written in our readme.

File: docs/reference/providers/backend.md (hunk 1 above)
Comment (rakshitgondwal) [reply docs/reference/providers/backend.md:6]: Sure, will do.

File: docs/reference/providers/backend.md (hunk 2 above)
Comment (AnaisUrlichs) [reply docs/reference/providers/backend.md:13]: @rakshitgondwal could you please highlight OpenAI as a default? :)

File: docs/reference/providers/backend.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (AnaisUrlichs) [reply docs/reference/providers/backend.md:46]: Awesome, thank you

File: docs/reference/providers/backend.md (hunk 2 above)
Comment (rakshitgondwal) [reply docs/reference/providers/backend.md:13]: Highlight as in? Should I write `OpenAI is the default backend for K8sGPT` in bold?

Comment (AnaisUrlichs) [reply docs/reference/providers/backend.md:13]: Maybe add

> OpenAI is the default backend for K8sGPT. We recommend using OpenAI first if you are new to K8sGPT.

What do you think?

**Review by AlexsJones — APPROVED** [review]

**Comment by rakshitgondwal** [comment]
Thank you for the merge!

**Pull request merged**